src/sentry/static/sentry/app/components/events/interfaces/spans/header.tsx

```diff
@@ -44,7 +44,28 @@ type PropType = {
   event: EventTransaction;
 };
 
-class TraceViewHeader extends React.Component<PropType> {
+type State = {
+  minimapWidth: number | undefined;
+};
+
+class TraceViewHeader extends React.Component<PropType, State> {
+  state: State = {
+    minimapWidth: undefined,
+  };
+
+  componentDidUpdate() {
+    const {minimapInteractiveRef} = this.props;
+    if (minimapInteractiveRef.current) {
+      const minimapWidth = minimapInteractiveRef.current.getBoundingClientRect().width;
+      if (minimapWidth !== this.state.minimapWidth) {
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({
+          minimapWidth,
+        });
```

Comment:
Is there no better way of doing this than calling `setState` inside of `componentDidUpdate`? 🤔

Reply:
This is legal to do and a common pattern: https://reactjs.org/docs/react-component.html#componentdidupdate

Reply:
No, I'm just wondering if there was some listener we can hook onto instead of this, but this is fine as is 😄